app/services/service_result.rb

```diff
@@ -185,6 +254,8 @@ def map
     end
   end
 
+  # @return [Array<Object, nil>] the {#result} wrapped in an Array if the Service call succeeds, or an empty Array if
+  #   the Service call fails.
```

Comment:
I find the `nil` disturbing. Sadly there is no way to indicate the size in yard notation.

```suggestion
  # @return [Array<Object>] the {#result} wrapped in an Array if the Service call succeeds, or an empty Array if
  #   the Service call fails.
```

Reply:
@cbliard `#to_a` wraps `#result`, which can return `Object` or `nil` (we don't currently enforce the presence of a result for a successful Service call[^1], although ofc in practice, result is usually non-nil on success).

After doing a bit of Googling, it looks like think there is a syntax for fixed-sized arrays/tuples, etc. BUT it looks like it's not possible to delimit multiple types ([see discussion here](https://github.com/lsegal/yard/discussions/1579)).

(I am guessing that `Array()` can be used to denote an empty Array, but haven't seen it used in the wild.)

```rb
  # @return [Array(Object), Array()] the {#result} wrapped in an Array if the Service call succeeds, or an empty Array if
```

[^1]: something to add to a list of possible enhancements?